api/commands/run.ts

```diff
@@ -96,107 +100,84 @@ export class Runner {
     return Promise.all(this.changeListeners.map(listener => listener(this.result)));
   }
 
-  private async loop(resolve: () => void, reject: (value: any) => void) {
-    const pendingActions = this.pendingActions;
-    this.pendingActions = [];
-
-    const allFinishedDeps = this.result.actions.map(action => action.name);
-    const allSuccessfulDeps = this.result.actions
-      .filter(
-        action =>
-          action.status === dataform.ActionExecutionStatus.SUCCESSFUL ||
-          action.status == dataform.ActionExecutionStatus.DISABLED
-      )
-      .map(fn => fn.name);
-
-    pendingActions.forEach(async action => {
-      const finishedDeps = action.dependencies.filter(d => allFinishedDeps.indexOf(d) >= 0);
-      const successfulDeps = action.dependencies.filter(d => allSuccessfulDeps.indexOf(d) >= 0);
-      if (!this.cancelled && successfulDeps.length == action.dependencies.length) {
-        // All required deps are completed, start this action.
-        this.executeAction(action);
-      } else if (this.cancelled || finishedDeps.length == action.dependencies.length) {
-        await this.triggerChange();
-        // All deps are finished but they weren't all successful, or the run was cancelled.
-        // skip this action.
-        this.result.actions.push({
-          name: action.name,
-          status: dataform.ActionExecutionStatus.SKIPPED,
-          deprecatedSkipped: true
-        });
-      } else {
-        this.pendingActions.push(action);
-      }
+  private async executeGraph() {
+    const executeAllActionsReadyForExecution = async () => {
```

Comment:
Doesn't look like this has any reason to be inside executeGraph - make it class level function?